docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,529 @@
+
+## Pros and Cons of the applying policies on custom namespaces
+
+* Good, because it allows using Kubernetes RBAC
+* Good, because it provides more predictable behaviour as namespace-scoped policy affects only workloads in the same
+  namespace
+* Good, because it provides more Kubernetes-native UX
+* Bad, because adds complexity for users
+    * what's the right namespace for the policy?
+    * how cross-policy refs work?
+    * how policy order works?
+
+## Pros and Cons of doing nothing
+
+* Good, because we have time to work on other features
+* Bad, because cluster operator has to grant write permissions to `kuma-system` namespace to everyone who works with
+  policies
+* Bad, because low isolation between teams (team-a can unintentionally break polices of the team-b)
+
+## Implementation
+
+Applying a policy on the custom namespace **affects requests to the pods in the custom namespace**.
+Since some policies are applied on the client-side, the scope of policy's effect is not limited to the namespace
+where it was applied. 
+For example, the following policy sets the timeout on outbounds of all consumers of the backend service:
+
+```yaml
+kind: MeshTimeout
+metadata:
+  namespace: backend-ns
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef: 
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+      default:
+        connectionTimeout: 5s
+```
+
+Important to note, even though policy in `backend-ns` changes proxy configurations in `frontend-ns` 
+the top-level targetRef still servers the purpose of showing what proxy configurations are going to change.
+Policy's namespace **does not** implicitly turn the top-level targetRef to `MeshSubset{tags:{"kuma.io/namespace":"backend-ns"}}`.
+
+### Namespace-scoped policies and multizone
+
+Namespace-scoped policies can be applied to a custom namespace only on zones as there are simply no DPPs and custom
+namespaces on Global.
+
+After a user applied a policy on zone's custom namespace, the webhook automatically adds a `k8s.kuma.io/namespace` label to the policy.
+The label is part of the hash suffix when policy is synced to global.
+
+As mentioned previously, a policy applied in producer namespace can affect envoy configs of consumers.
+Consumers can reside not only in different namespaces, but also in different zones.
+That's why, the policy applied in `backend-ns` in `zone-1` has to be synced to global and then to other zones.
+With the increased blast radius of the policies comes great responsibility, 
+we don't want small mistake in the policy to break traffic on the other side of the world.
+A few things could be done to protect users from impactful mistakes:
+* Enhanced validation. If policy is created in `backend-ns` of the `zone-1` and the top-level targetRef is `Mesh`, 
+then `to[].targetRef` should have both `k8s.kuma.io/namespace: backend-ns` and `kuma.io/zone: zone-1`. 
+* Ability to opt-in or opt-out from syncing the policy to other zones.
```

Comment:
I don't think that's necessary you can always use a top `spec.tagerRef.kind: MeshSubset` to reduce blast radius no? That's what it is for no?

Reply:
It probably wasn't clear that these points:

```yaml
* Enhanced validation...
* Ability to opt-in or opt-out...
```

are followed by the separate sections to elaborate on each point:

```md
#### Enhanced validation
...
#### Considered options on opting-in(out) to cross-zone policy syncing
...
```

and the decision for opting-in(out) is exactly what you said – we can always use a top `spec.targetRef.kind: MeshSubset` to reduce blast radius.